integrations/pagerduty/main.py

```diff
@@ -0,0 +1,49 @@
+from typing import Any, Dict, List
+from clients.pagerduty import PagerDutyClient
+from clients.utils import ObjectKind
+from port_ocean.context.ocean import ocean
+from starlette.requests import Request
+
+pager_duty_client = PagerDutyClient(
+    ocean.integration_config["token"],
+    ocean.integration_config["url"],
+    ocean.integration_config["app_url"],
+)
+
+
+@ocean.on_resync(ObjectKind.INCIDENTS)
+async def on_incidents_resync(kind: str) -> List[Dict[Any, Any]]:
+    incidents = pager_duty_client.paginate_request_to_pager_duty(data_key="incidents")
+    return incidents
+
+
+@ocean.on_resync(ObjectKind.SERVICES)
+async def on_services_resync(kind: str) -> List[Dict[Any, Any]]:
+    services = pager_duty_client.paginate_request_to_pager_duty(data_key="services")
+    return services
+
+
+@ocean.router.post("/webhook")
+async def upsert_incident_webhook_handler(
+    data: Dict[str, Any], request: Request
+) -> None:
+    if data["event"]["event_type"] in pager_duty_client.service_delete_events:
+        await ocean.unregister_raw("services", [data["event"]["data"]])
+
+    elif data["event"]["event_type"] in pager_duty_client.incident_upsert_events:
+        event_data = data["event"]["data"]
+        event_data["assignments"] = [{"assignee": event_data["assignees"][0]}]
```

Comment:
why are you adding more information to the evnet?

Reply:
This ensures the webhook event payload is identical to the API response when we sync. 

Here's a sample webhook event payload for incidents. examine the `assignees` property
![image](https://github.com/port-labs/Port-Ocean/assets/15999660/59992616-59c7-4492-bd1f-23b9964ac44f)

Below is also the response from the API when we sync. Again, you can see that the same `assignee` object is encapsulated under `assignments`
![image](https://github.com/port-labs/Port-Ocean/assets/15999660/de0362cb-a4eb-4426-8228-414b39a7b763)

Reply:
We usually request the api instea of using the event payload

Reply:
That makes sense, then.